Samples/ExtendedControlAndMetadata/EyeGazeAndBackgroundSegmentation/README.md

```diff
@@ -0,0 +1,245 @@
+
+```cpp
+// PropertyInquiry.cpp
+
+winrt::CameraKsPropertyHelper::IMetadataPayload PropertyInquiry::ExtractFrameMetadata(winrt::Windows::Media::Capture::Frames::MediaFrameReference const& frame, winrt::CameraKsPropertyHelper::FrameMetadataKind const& metadataKind)
+    {
+        static guid captureMetadataGuid = guid(0x2EBE23A8, 0xFAF5, 0x444A, { 0xA6, 0xA2, 0xEB, 0x81, 0x08, 0x80, 0xAB, 0x5D }); // MFSampleExtension_CaptureMetadata
+        static guid backgroundSegmentationMaskGuid = guid(0x3f14dd3, 0x75dd, 0x433a, { 0xa8, 0xe2, 0x1e, 0x3f, 0x5f, 0x2a, 0x50, 0xa0 }); // MF_CAPTURE_METADATA_FRAME_BACKGROUND_MASK
+        static guid dwKey = guid(0x276f72a2, 0x59c8, 0x4f69, { 0x97, 0xb4, 0x6, 0x8b, 0x8c, 0xe, 0xc0, 0x44 });
+        CameraKsPropertyHelper::IMetadataPayload result = nullptr;
+
+        if (frame == nullptr)
+        {
+            throw hresult_invalid_argument(L"null frame");
+        }
+        auto properties = frame.Properties();
+        if (properties != nullptr)
+        {
+            if (metadataKind == FrameMetadataKind::MetadataId_BackgroundSegmentationMask)
+            {
+                // 1. look for the presence of capture metadata sub-map
+                auto captureMetadata = properties.TryLookup(captureMetadataGuid);
+                if (captureMetadata != nullptr)
+                {
+                    auto captureMetadataPropSet = captureMetadata.as<Windows::Foundation::Collections::IMapView<guid, winrt::Windows::Foundation::IInspectable>>();
+
+                    // 2. look for the presence of the specific capture metadata we want
+                    auto metadata = captureMetadataPropSet.TryLookup(backgroundSegmentationMaskGuid);
+
+                    if (metadata != nullptr)
+                    {
+                        auto metadataPropVal = metadata.as<Windows::Foundation::IPropertyValue>();
+
+                        // 3. deserialize the IPropertyValue we retrieve
+                        result = make<BackgroundSegmentationMaskMetadata>(metadataPropVal);
+                    }
+                }
+                else
+                {
+                    // no metadata present
+                    return result;
+                }
+            }
+            else
+            {
+                throw hresult_not_implemented(L"no handler for FrameMetadataKind = " + to_hstring((int)metadataKind));
+            }
+        }
+        // else there is no frame metadata present
+
+        return result;
+    }
+```
+
+## Eye Gaze Correction
+the eye gaze correction DDI (*```KSPROPERTY_CAMERACONTROL_EXTENDED_EYEGAZECORRECTION```*) has been introduced in Windows strating with build 20348. If supported by the driver, it allows to slighly change the predominant face's eye position so that the main subject appears as if looking directly into the camera.
```

Comment:
>allows to slighly change [](http://example.com/codeflow?start=179&length=24)

"allows to slightly change" -> ""slightly changes"

Reply:
good catch, I did not mean to allow to slyly change the eye gaze position